blockchain/vm/evm.go

```diff
@@ -538,6 +538,33 @@ func (evm *EVM) CreateWithAddress(caller types.ContractRef, code []byte, gas uin
 	return evm.create(caller, codeAndHash, gas, value, contractAddr, humanReadable, codeFormat)
 }
 
+func (evm *EVM) CurrentCodeFormat() params.CodeFormat {
+	switch {
+	case evm.chainRules.IsIstanbul:
+		return params.CodeFormatEVM2
+	default:
+		return params.CodeFormatEVM
+	}
+}
+
+func (evm *EVM) GetPrecompiledContract(caller, address common.Address) map[common.Address]PrecompiledContract {
+	switch {
+	case evm.chainRules.IsIstanbul:
+		// Below code is added to make vmLog(0x09), feePayer(0x0a), validateSender(0x0b) precompiled contract
+		// works after the istanbul incompatible change
+		// (1) check whether the contract(caller) is deployed before istanbul incompatible change
+		// (2) then, check whether the precompiled contract(address) is 0x09, 0x0a, 0x0b
+		// (3) if it is, use old precompiled contract set
```

Comment:
```suggestion
		// if contract(caller) is deployed before istanbul and 
		// (1) contract address is 0x09, 0x0a, 0x0b, use the old  precompiled contract set
		// (2) contract address is 0x06, 0x07, 0x08, use the new  precompiled contract set
		//       (the only difference is gas cost)
```

Reply:
Thanks for revising comments, it's really helpful. Reflected at https://github.com/klaytn/klaytn/pull/912/commits/1c0928ba549158af0d357eed0e6d0a7763e14bd8.